lib/dor/text_extraction/abbyy/results.rb

```diff
@@ -5,27 +5,35 @@ module TextExtraction
     module Abbyy
       # Parses .xml.result.xml file that gets created when Abbyy sees a XMLTICKET
       class Results
-        attr_reader :result_path, :druid, :run_index
+        attr_reader :result_path, :druid, :run_index, :software_name, :software_version
 
         # Naming convention for result XML files:
         # ab123cd4567.xml.result.xml
         # ab123cd4567.xml.0001.result.xml
         RESULT_FILE_PATTERN = /(?<druid>\w+)\.xml(?:\.(?<run_index>\d+))?\.result\.xml/
 
+        # Class variables so we can use the last value if we don't have ALTO to check
+        @last_software_name = nil
+        @last_software_version = nil
+        class << self
+          attr_accessor :last_software_name, :last_software_version
+        end
+
         def initialize(result_path:)
           @result_path = result_path
           druid, run_index = RESULT_FILE_PATTERN.match(File.basename(result_path)).captures
           @druid = druid
           @run_index = (run_index || 0).to_i
+          @software_name = processing_metadata[:software_name] || self.class.last_software_name
+          @software_version = processing_metadata[:software_version] || self.class.last_software_version
```

Comment:
removed this in response to slack discussion: we now only get the version if it has ALTO, and for the software name I just hardcoded a default in case of PDF.